backend/oasst_backend/task_repository.py

```diff
@@ -128,6 +129,7 @@ def store_task(
         assert task_model.id == task.id
         return task_model
 
+    @managed_tx_method(CommitMode.COMMIT)
```

Comment:
many of those methods are now still Commit .. shouldn't they all become FLUSH and we only commit in the outer transaction, e.g. in the controller where the web call is handled?

Reply:
I have done the same. Added to FLUSH to more atomic functions(i.e functions used by others) and COMMIT to outer functions.

This looked to be in the outer transaction. that way this logic of managed transaction is encapsulated inside the pr (PromptRepository) and the web call handler need not know of the db tx details
e.g 
pr.task_repository.bind_frontend_message_id(task_id=task_id, frontend_message_id=ack_request.message_id)

@router.post("/{task_id}/ack", response_model=None, status_code=HTTP_204_NO_CONTENT)